frontend/src/utils/DeviceStore/deviceStore.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * Device store that retrieves the stored device ID for a given device type (audio or video)
+ * and checks if it is still connected.
+ */
+export default class DeviceStore {
+  private devices: MediaDeviceInfo[] = [];
+
+  async init() {
+    if (navigator.mediaDevices) {
+      this.devices = await navigator.mediaDevices.enumerateDevices();
+    }
+  }
+
+  getConnectedDeviceId(kind: 'audioinput' | 'videoinput'): string | undefined {
+    const key = kind === 'videoinput' ? 'videoSource' : 'audioSource';
+    const storedId = window.localStorage.getItem(key);
```

Comment:
This can be updated to use your storage methods and keys

Reply:
oh yes, definitely, I forgot to do that 😅